python/nano/src/bigdl/nano/pytorch/dispatcher.py

```diff
@@ -0,0 +1,73 @@
+#
+# Copyright 2016 The BigDL Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from importlib.util import find_spec
+
+
+_mapping_torch = None
+
+
+def _get_patch_map():
+    global _mapping_torch
+
+    # decide if generate
+    patch_torch = find_spec("pytorch_lightning") is not None
```

Comment:
I think you can still patch `torchvision` even if `pytorch_lightning` is not available?

Reply:
yeah, I can do that for the patching logic.

btw, I don't think a user who install `torchvision` but not `pytorch_lightning` can use nano smoothly (due to some `__init__.py`'s importing logic), we may fix those importing in later PR if necessary.

Reply:
> yeah, I can do that for the patching logic.
> 
> btw, I don't think a user who install `torchvision` but not `pytorch_lightning` can use nano smoothly (due to some `__init__.py`'s importing logic), we may fix those importing in later PR if necessary.

The goal is to support pytorch-only users using `TorchNano`; we should not depend on `pytorch_lightning` in this case.

Reply:
yeah, but `TorchNano` is inherited from `LightningLite` which is depended on `pytorch_lightning`. So maybe the only goal for this is to support those who only want to use `torchvision`'s acceleration.